workflow/executor.go

```diff
@@ -319,40 +319,44 @@ func applyLifecycleScopes(
 ) error {
 	allNamespaces := make(map[string]schema.Scope)
 	for workflowStepID, stepLifecycle := range stepLifecycles {
-		// The format will be $.steps.step_name.stage.<input|outputs.output_id>
-		// First, get the input schema and output schemas, and apply them.
 		for _, stage := range stepLifecycle.Stages {
 			prefix := "$.steps." + workflowStepID + "." + stage.ID + "."
 			// Apply inputs
-			// Example: $.steps.wait_step.starting.input
+			// Example: $.steps.wait_step.starting.inputs.
 			addInputNamespacedScopes(allNamespaces, stage, prefix+"inputs.")
 			// Apply outputs
-			// Example: $.steps.wait_step.outputs.success
+			// Example: $.steps.wait_step.outputs.outputs.success
 			addOutputNamespacedScopes(allNamespaces, stage, prefix+"outputs.")
```

Comment:
Given that `prefix+"inputs."` produces `$.steps.wait_step.starting.inputs.` at line 325, how does `prefix+"outputs."` produce `$.steps.wait_step.outputs.outputs.success` instead of `$.steps.wait_step.starting.outputs.` at line 328?

In the example, `stage.ID` appears to be `starting` (not `outputs`), and `success` doesn't appear in the parameter (although presumably it gets added inside the function call).

Reply:
In that case, the stage ID is `outputs`. Outputs doesn't have any inputs with scopes, and starting doesn't have any outputs with scopes, so I needed to use different stages in the examples.
I'll clarify the stage ID.